test/functional/http/post/5.dapps.js

```diff
@@ -0,0 +1,503 @@
+					// badTransactions.push(transaction);
+				})
+				.then(function (res) {
+					// Transaction with same info but different ID (due to timeOffSet parameter)
+					transaction = node.lisk.dapp.createDapp(account.password, null, dappDuplicate, 1);
+
+					return sendTransactionPromise(transaction);
+				})
+				.then(function (res) {
+					node.expect(res).to.have.property('success').to.be.ok;
+					node.expect(res).to.have.property('transactionId').to.equal(transaction.id);
+					// TODO: Enable when transaction pool order is fixed
+					// goodTransactions.push(transaction);
+				});
+		});
+
+		it('two different dapps with same name should be ok and only last transaction to arrive will be confirmed', function () {
+			transaction = node.lisk.dapp.createDapp(account.password, null, dappDuplicateNameFail);
+
+			return sendTransactionPromise(transaction)
+				.then(function (res) {
+					node.expect(res).to.have.property('success').to.be.ok;
+					node.expect(res).to.have.property('transactionId').to.equal(transaction.id);
+					// TODO: Enable when transaction pool order is fixed
+					// badTransactions.push(transaction);
+				})
+				.then(function (res) {
+					// Transaction with same info but different ID (due to timeOffSet parameter)
+					transaction = node.lisk.dapp.createDapp(account.password, null, dappDuplicateNameSuccess);
+
+					return sendTransactionPromise(transaction);
+				})
+				.then(function (res) {
+					node.expect(res).to.have.property('success').to.be.ok;
+					node.expect(res).to.have.property('transactionId').to.equal(transaction.id);
+					// TODO: Enable when transaction pool order is fixed
+					// goodTransactions.push(transaction);
+				});
+		});
+
+		it('two different dapps with same link should be ok and only last transaction to arrive will be confirmed', function () {
+			transaction = node.lisk.dapp.createDapp(account.password, null, dappDuplicateLinkFail);
+
+			return sendTransactionPromise(transaction)
+				.then(function (res) {
+					node.expect(res).to.have.property('success').to.be.ok;
+					node.expect(res).to.have.property('transactionId').to.equal(transaction.id);
+					// TODO: Enable when transaction pool order is fixed
+					// badTransactions.push(transaction);
+				})
+				.then(function (res) {
+					// Transaction with same info but different ID (due to timeOffSet parameter)
+					transaction = node.lisk.dapp.createDapp(account.password, null, dappDuplicateLinkSuccess);
+
+					return sendTransactionPromise(transaction);
+				})
+				.then(function (res) {
+					node.expect(res).to.have.property('success').to.be.ok;
+					node.expect(res).to.have.property('transactionId').to.equal(transaction.id);
+					// TODO: Enable when transaction pool order is fixed
+					// goodTransactions.push(transaction);
+				});
+		});
+	});
+	
+	describe('confirmation', function () {
+
+		shared.confirmationPhase(goodTransactions, badTransactions);
+	});
+
+	describe('validation', function () {
```

Comment:
Any other fields that cannot be duplicated?

Reply:
The code is checking just the two of them https://github.com/LiskHQ/lisk/blob/bd22597b26a0e36a32232f2d92737a2905fe9a31/logic/dapp.js#L163
Actually I don't get that last else.

Reply:
https://github.com/LiskHQ/lisk/blob/bd22597b26a0e36a32232f2d92737a2905fe9a31/logic/dapp.js#L151
So name, link and transactionId - add a test for to check if it is impossible to register exactly the same transaction twice